airflow/api_fastapi/core_api/routes/public/task_instances.py

```diff
@@ -331,3 +333,38 @@ def get_task_instances(
         ],
         total_entries=total_entries,
     )
+
+
+@task_instances_router.get(
+    "/{task_id}/tries/{task_try_number}",
+    responses=create_openapi_http_exception_doc([status.HTTP_404_NOT_FOUND]),
+)
+def get_task_instance_try_details(
+    dag_id: str,
+    dag_run_id: str,
+    task_id: str,
+    task_try_number: int,
+    session: Annotated[Session, Depends(get_session)],
+    map_index: int = -1,
+) -> TaskInstanceHistoryResponse:
+    """Get task instance details by try number."""
+
+    def _query(TI):
```

Comment:
```suggestion
    def _query(TI):
```
```suggestion
    def _query(orm_object: Base):
```
`orm_object` is more explicit.

Reply:
Also you can add typing (param and return type) Select